routers/api/v1/repo/hook.go

```diff
@@ -153,6 +154,17 @@ func TestHook(ctx *context.APIContext) {
 		return
 	}
 
+	ref := git.BranchPrefix + ctx.Repo.Repository.DefaultBranch
+	if refName := ctx.FormTrim("ref"); refName != "" {
+		if ctx.Repo.GitRepo.IsBranchExist(refName) {
+			ref = git.BranchPrefix + refName
+		} else if ctx.Repo.GitRepo.IsTagExist(refName) {
+			ref = git.TagPrefix + refName
+		}
+		// If the ref is a commit id, it's not worth finding out all branches contain the commit
+		// and picking the most likely one, so just use the default branch for now.
+	}
```

Comment:
IIRC, it's suggested to pass accurate ref name instead of guessing about the ref type @zeripath https://github.com/go-gitea/gitea/pull/20357#issuecomment-1184307624

Reply:
👍🏻I see, resolved it.